core-tests/shared/src/test/scala/zio/ZScheduleSpec.scala

```diff
@@ -294,6 +307,36 @@ object ZScheduleSpecUtil {
       res <- ref.update(_ + 1).repeat(schedule)
     } yield res
 
+  /**
+   * Run a schedule using the provided input and collect all outputs
+   */
+  def run[A, B](
```

Comment:
It seems like we should 't need this amount of ceremony anymore after TestClock, TestRandom etc.

Reply:
Yep, this can be trivialized with fiberTime. Working on it as we speak :grin: